internal/renderers/skia/textlayout.rs

```diff
@@ -212,6 +212,23 @@ pub fn cursor_rect(
         );
     }
 
+    // SkParagraph::getRectsForRange() does not report the text box of a trailing newline
+    // correctly. Use the last line's metrics to get the correct coordinates (#3590).
+    if cursor_pos == string.len() && string.ends_with('\n') {
```

Comment:
This might also need checking for '\u{2028}' and '\u{2029}', as those also cause newlines.

Reply:
Nice, thanks! Both had the same issue.